### PR TITLE
wallet: remove obsolete peers.

### PR DESCRIPTION
**Pull request by rustyrussell**
Fixes: #1029
Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>

### CODE REVIEW

**Comment by ZmnSCPxj** [comment]
ACK bd853d53980663dabea3e21057f058561220e915

File: wallet/db.c [hunk 1]
```diff
@@ -196,6 +196,8 @@ char *dbmigrations[] = {
     "DELETE FROM channels WHERE state=1",
     /* Keep track of db ugprades, for debugging */
     "CREATE TABLE db_upgrades (upgrade_from INTEGER, lightning_version TEXT);",
+    /* We didn't used to clean up peers when their channels were gone. */
```
Comment (practicalswift) [inline wallet/db.c]: Should it be "We didn't use to …"? :-)

Comment (rustyrussell) [reply wallet/db.c]: Apparently you're right!  Will fix...

**Comment by cdecker** [comment]
ACK 67ca2a9

**Pull request merged**